### PR TITLE
BRMP 10103 and BRMP 10201 - Acquisition Resources Page

### PR DESCRIPTION
**Pull request by AubreyTharpe**
https://cm-jira.usa.gov/browse/BRMP-10103
https://cm-jira.usa.gov/browse/BRMP-10201

Changes
----
- Refactored API changes to reflect the current drupal
- Added value mapping in the service layer for acquisition vehicles
- Refactored frontend with mapped values
- Added new drupal fields from BRMP-10103 to page
- Added notices on acquisition vehicle display

### CODE REVIEW

File: src/components/features/acquisitions/acquisitionResources.js [hunk 1]
```diff
@@ -335,50 +357,56 @@ export default class AcquisitionResources extends React.Component {
...
-                                            <ExternalLink href={bpa.attributes.field_bpa_ordering_requirements.uri}>
-                                                SIISS Ordering Guide {/* TODO: remove this */}
-                                            </ExternalLink>
-                                        </div>
+                                        bpa.orderingRequirements.title ? (
+                                            <div>
+                                                <div>{bpa.orderingRequirements.title}</div>
+                                                <br />
+                                                <ExternalLink href={bpa.orderingRequirements.uri}>
+                                                    SIISS Ordering Guide {/* TODO: remove this */}
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResources.js:365]: need to remove

File: src/components/features/acquisitions/acquisitionResources.js [hunk 2]
```diff
@@ -25,19 +24,26 @@ export default class AcquisitionResources extends React.Component {
...
-
-        getBlanketPurchaseAgreement().then(blanketPurchaseAgreements => {
-            this.setState({ blanketPurchaseAgreements })
-        });
-
     }
 
+    renderBPAAcquisitionVehicles(acquisitionVehicles) {
+        return acquisitionVehicles.map(av => {
+            return (
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResources.js:37]: return not necessary

File: src/components/features/acquisitions/acquisitionResources.js [hunk 3]
```diff
@@ -154,12 +177,12 @@ export default class AcquisitionResources extends React.Component {
                                     </td>
                                     <td>
                                         {
-                                            av.attributes.field_av_ordering_activities.map((oa, i) => {
+                                            av.orderingActivities.map((oa, i) => {
                                                 return (<div key={i}>
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResources.js:181]: return

File: src/components/features/acquisitions/acquisitionResources.js [hunk 4]
```diff
@@ -243,8 +266,8 @@ export default class AcquisitionResources extends React.Component {
                                     </td>
                                     <td>
                                         {
-                                            av.relationships.field_av_socioeconomics.data.map((se, i) => {
-                                                return this.state.socioeconomics[se.id] // use id of socioeconomic to display name
+                                            av.socioeconomics.map((se, i) => {
+                                                return this.state.socioeconomics[se] // use id of socioeconomic to display name
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResources.js:270]: return

File: src/components/features/acquisitions/acquisitionResources.js [hunk 5]
```diff
@@ -335,50 +357,56 @@ export default class AcquisitionResources extends React.Component {
...
                                 </td>
                             </tr>
                             <tr>
                                 <td>Socioeconomic:</td>
                                 <td>                              
                                     {
-                                            bpa.relationships.field_bpa_socioeconomics.data.map((se, i) => {
-                                                return this.state.socioeconomics[se.id] // use id of socioeconomic to display name
+                                            bpa.socioeconomics.map((se, i) => {
+                                                return this.state.socioeconomics[se] // use id of socioeconomic to display name
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResources.js:409]: return

File: src/components/features/acquisitions/acquisitionResourcesService.js [hunk 6]
```diff
@@ -20,21 +20,105 @@ const mapGovtCspsFields = (csps) => {
...
 }
 
-export const getBlanketPurchaseAgreement = () => {
-    return getGenericBlanketPurchaseAgreement().then(data => {
-        return Promise.resolve(data);
-    });
+const mapAVTaxonomy = (taxonomies) => {
+  const mapped = {};
+  taxonomies.forEach(taxonomy => {
+    if(taxonomy.uuid){
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResourcesService.js:31]: mapped[taxonomy.uuid ? taxonomy.uuid : taxonomy.id] = {...taxonomy}

File: src/components/features/acquisitions/acquisitionResourcesService.js [hunk 7]
```diff
@@ -20,21 +20,105 @@ const mapGovtCspsFields = (csps) => {
...
+            field_av_order_types:             orderTypes,
+            field_av_scope:                   scope,
+            field_av_term_date:               termDate,
+            field_av_type:                    type,
+            field_av_website:                 website,
+            field_other_than_small_business:  otherThanSmallBusiness
+        }} = av;
+        const socioeconomics = s?.data.map(cat => cat.id) || [];
+        const listAVs = a.data.map(av => {
+          return {
```
Comment (snoopysnake) [inline src/components/features/acquisitions/acquisitionResourcesService.js:76]: return

**Review by snoopysnake — APPROVED** [review]
ok

File: src/components/features/acquisitions/acquisitionResources.js (hunk 1 above)
Comment (AubreyTharpe) [reply src/components/features/acquisitions/acquisitionResources.js:365]: Keeping in for reminder of future work.

**Pull request merged**